flask_jwt_extended/utils.py

```diff
@@ -76,14 +76,21 @@ def decode_token(encoded_token, csrf_value=None):
     unverified_claims = jwt.decode(
         encoded_token, verify=False, algorithms=config.algorithm
     )
-    secret = jwt_manager._decode_key_callback(unverified_claims)
+    unverified_headers = jwt.get_unverified_header(encoded_token)
+    # Attempt to call callback with both claims and headers, but fallback to just claims
+    # for backwards compatibility
+    try:
+        secret = jwt_manager._decode_key_callback(unverified_claims, unverified_headers)
+    except TypeError:
+        secret = jwt_manager._decode_key_callback(unverified_claims)
```

Comment:
Lets issue a depreciation warning here as well. An example of how that looks can be seen here: https://github.com/vimalloc/flask-jwt-extended/blob/master/flask_jwt_extended/config.py#L165